plugin/pkg/admission/security/podsecuritypolicy/admission.go

```diff
@@ -287,20 +303,47 @@ func (c *podSecurityPolicyPlugin) createProvidersFromPolicies(psps []*extensions
 
 // getMatchingPolicies returns policies from the store.  For now this returns everything
 // in the future it can filter based on UserInfo and permissions.
-func getMatchingPolicies(store cache.Store, user user.Info, sa user.Info) ([]*extensions.PodSecurityPolicy, error) {
+func getMatchingPolicies(store cache.Store, user user.Info, sa user.Info, authz authorizer.Authorizer) ([]*extensions.PodSecurityPolicy, error) {
 	matchedPolicies := make([]*extensions.PodSecurityPolicy, 0)
 
 	for _, c := range store.List() {
 		constraint, ok := c.(*extensions.PodSecurityPolicy)
 		if !ok {
 			return nil, errors.NewInternalError(fmt.Errorf("error converting object from store to a pod security policy: %v", c))
 		}
-		matchedPolicies = append(matchedPolicies, constraint)
+
+		if authorizedForPolicy(user, constraint, authz) || authorizedForPolicy(sa, constraint, authz) {
+			matchedPolicies = append(matchedPolicies, constraint)
+		}
 	}
 
 	return matchedPolicies, nil
 }
 
+// authorizedForPolicy returns true if info is authorized to perform a "get" on policy.
+func authorizedForPolicy(info user.Info, policy *extensions.PodSecurityPolicy, authz authorizer.Authorizer) bool {
+	if info == nil {
+		return false
+	}
+	attr := buildAttributes(info, policy)
+	allowed, _, _ := authz.Authorize(attr)
+	return allowed
+}
+
+// buildAttributes builds an attributes record for a SAR based on the user info and policy.
+func buildAttributes(info user.Info, policy *extensions.PodSecurityPolicy) authorizer.Attributes {
+	attr := authorizer.AttributesRecord{
+		User:            info,
+		Verb:            "use",
+		Namespace:       policy.Namespace,
```

Comment:
this is interesting but a fundamental change.  I can definitely see the benefit of allowing ns admins to create more restrictive PSPs and granting access there but it would require some other changes and maybe some discussion on whether folks actually want to admin policy this way.  Will take a note to discuss